horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -1004,4 +1012,213 @@ public DatasetList listDatasetsBySchema(String uri, Integer limit, Integer page,
+                              if (result != null) {
+                                 logTransformationMessage(test, run, TransformationLog.ERROR, "transformation "+t.name+" function", result);
+                                 functionError = true;
+                              }
+                           } finally {
+                              context.leave();
+                              if (out.size() > 0) {
+                                 logTransformationMessage(test, run, TransformationLog.ERROR, "transformation "+t.name+" function", result);
+                              }
+                           }
+                        }
+                        if (result != null) {
+                           JsonNode node = Util.toJsonNode(result);
+                           if (node == null || functionError) {
+                              log.errorf("Conversion of Transformation "+t.name+" function result failed. See prior logs for cause.");
+                              node = instance.objectNode();
+                              transformedSingleNodeResults.add(node);
+                           } else if (node.isArray() ) {
+                              // let's assume (always dangerous) only 1 Transformer produces an array
+                              if (containsArray) { // if not then log a warning and overwrite the prior
+                                 log.warnf("Detected transformation issue that multiple arrays produced by functions. Prior array is being dropped %s", transformedArrayNodeResult);
+                              }
+                              containsArray = true;
+                              transformedArrayNodeResult = node;
+                              arrayPosition = transformedSingleNodeResults.size();
+                              transformedSingleNodeResults.add(null);
+                           } else {
+                              transformedSingleNodeResults.add(node);
+                           }
+                        }
+                     }
+                  }
+                  extractorObjectsPerTrans = null;
+                  int max = 0;
+                  if (null != transformedArrayNodeResult) {
+                     max = transformedArrayNodeResult.size();
+                  } else {
+                     max = transformedSingleNodeResults.size();
+                  }
+                  ArrayNode data = instance.arrayNode();
+                  for (int count = 0 ; count < max; count += 1 ) {
+                     if (count == arrayPosition) {
+                        data.add(transformedArrayNodeResult.get(count));
+                        // bypass the single node result element. which is null
+                     } else {
+                        data.add(transformedSingleNodeResults.get(count));
+                     }
+                  }
+                  dataSets.add(new DataSet(run.start, run.stop
```

Comment:
I might got lost in the indents, but you should **not** produce DataSet for each schema.

Reply:
ok